airflow/utils/log/file_task_handler.py

```diff
@@ -179,6 +179,8 @@ class FileTaskHandler(logging.Handler):
     inherits_from_empty_operator_log_message = (
         "Operator inherits from empty operator and thus does not have logs"
     )
+    executor_instances: dict[str, BaseExecutor] = {}
+    default_executor_key = "_default_executor"
```

Comment:
You could just allow `None` as a key type with an or, but this works as well :+1: